### ISSUE TITLE
Longhorn volume access issue after rancher node restart

### ISSUE THREAD

**Issue by AmitDJagtap** [rancher/longhorn#226]
I have a Rancher Kubernetes Cluster I am working with longhorn v0.3-rc4. I am facing the following issue.
-  Add the bitnami library to Rancher Catalog
-  Create a MySQL cluster from bitnami chart with master and 2 salves
-  Set the persitance to enabled in Chart answers
  ```
      master.persistence.enabled: true
      master.persistence.storageClass: longhorn
      master.persistence.size: 50Gi

     slave.persistence.enabled: true
     slave.persistence.storageClass: longhorn
     slave.persistence.size: 50Gi
   ```
-  The cluster starts and PVC are bound and healthy. All Good
-  Now Since I am testing for a VM failover recovery solution, I restart all the VM's or Nodes in my RKE cluster.
-  After restart the SQL cluster does not start. This is what happens
      1. Pods get scheduled and start
      2. Mount volume succeeds ( PVC get attached after a restart, works now :thumbsup:)
      3. But I get a permission issue for MySQL user.
Below are the Rancher events and Container logs from MySQL pod.

RANCHER EVENTS
___________________
``` 
Normal    Pulling    pulling image "docker.io/bitnami/mysql:5.7"    15 minutes ago
Normal    Created    Created container    15 minutes ago
Normal    Started    Started container    15 minutes ago
Normal    Pulled    Successfully pulled image "docker.io/bitnami/mysql:5.7"    15 minutes ago
Normal    Scheduled    Successfully assigned mysql-mysql-master-0 to dev-clstr-2-2    13 minutes ago
Normal    SuccessfulMountVolume    MountVolume.SetUp succeeded for volume "custom-init-scripts"    13 minutes ago
Normal    SuccessfulMountVolume    MountVolume.SetUp succeeded for volume "config"    13 minutes ago
Normal    SuccessfulMountVolume    MountVolume.SetUp succeeded for volume "default-token-s7fkz"    13 minutes ago
Normal    SuccessfulMountVolume    MountVolume.SetUp succeeded for volume "pvc-21372ff0-9f9d-11e8-9d0b-0250e400d596"    13 minutes ago
Normal    Pulling    pulling image "docker.io/bitnami/mysql:5.7"    12 minutes ago
Normal    Pulled    Successfully pulled image "docker.io/bitnami/mysql:5.7"    12 minutes ago
Normal    Created    Created container    12 minutes ago
Normal    Started    Started container    12 minutes ago
Warning    BackOff    Back-off restarting failed container    3 minutes ago
```

LOGS
________
```
Welcome to the Bitnami mysql container
Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-mysql
Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-mysql/issues
nami    INFO  Initializing mysql
mysql   INFO
mysql   INFO  ########################################################################
mysql   INFO   Installation parameters for mysql:
mysql   INFO     Persisted data and properties have been restored.
mysql   INFO     Any input specified will not take effect.
mysql   INFO   This installation requires no credentials.
mysql   INFO  ########################################################################
mysql   INFO
nami    INFO  mysql successfully initialized
INFO  ==> Starting mysql...
INFO  ==> Starting mysqld_safe...
2018-08-16T09:42:03.859253Z mysqld_safe error: log-error set to '/opt/bitnami/mysql/logs/mysqld.log', however file don't exists. Create writable for user 'mysql'.
```
**Other Issues & Observations**
1. In the longhorn UI, the volume get attached and is healthy after node restart. But When I select any particular volume , I get and error - 
```
fail to list snapshot: error listing snapshot: Failed to execute: /var/lib/rancher/longhorn/engine-binaries/rancher-longhorn-engine-v0.3-rc4/longhorn [--url http://10.42.1.199:9501 snapshot info], output time="2018-08-16T13:06:56Z" level=fatal msg="Error running snapshot info command: BUG: Inconsistent snapshot info: {Name:volume-head Parent:b568bfdd-b734-403f-818a-f90e852a1bea Children:map[] Removed:false UserCreated:false Created:2018-08-16T12:57:06Z Size:344064 Labels:map[]} vs {Name:volume-head Parent:b568bfdd-b734-403f-818a-f90e852a1bea Children:map[] Removed:false UserCreated:false Created:2018-08-16T12:57:06Z Size:0 Labels:map[]}" , error exit status 1
```
2. Thus I am also not able to view my snapshots and backup which I took before restart.
3. On the backups tab I can view the backups , when I select one of the bacups and click restore the new volume does not get attached and is stuck in attaching state.

**Comment by yasker** [comment]
We would try to reproduce this issue.

But the restore backup one sounds a little weird. The newly created volume from backup should be in `detached` state. So you have attached it manually?

**Comment by yasker** [comment]
@AmitDJagtap Sounds like you're using the same volume name? In that case, it may be auto attached if Flexvolume driver is used. Let's reproduce the issue in our lab.

**Comment by yasker** [comment]
Should be fixed in v0.3.1 release with CSI drivers. https://github.com/rancher/longhorn/releases/tag/v0.3.1
